src/rules/fileHeaderRule.ts

```diff
@@ -79,6 +89,18 @@ export class Rule extends Lint.Rules.AbstractRule {
         return [];
     }
 
+    private doesNewLineEndingViolationExist(text: string, offset: number): boolean {
+        let violationExists = false;
+        const entireComment = ts.forEachLeadingCommentRange(
+            text,
+            offset,
+            (pos, end) => text.substring(pos, end + 2));
+
+        violationExists = entireComment !== undefined && entireComment.match(/^.*((\r)?\n){2,}$/mg) !== null;
```

Comment:
Maybe assign the regex to a constant w/ a more descriptive name and use `DESCRIPTIVE_NAME.test(entireComment)` instead of match?

Reply:
After testing this expression [here](https://regexr.com/), it seems it is not returning a reliable result. Maybe you could strip all spaces from `entireComment` and use `entireComment.lastIndexOf("\n") !== entireComment.length - 1`?

Reply:
Also looks like this regular expression is not returning reliable results (I tested [here](https://regexr.com/).